src/main/java/com/training/web/pages/flipkart/ProductPage.java

```diff
@@ -0,0 +1,56 @@
+package com.training.web.pages.flipkart;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ProductPage {
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    private WebDriver driver;
+    private WebDriverWait wait;
+
+    private By productName1 = By.xpath("//*[@class='G6XhRU']");
+    private By name = By.className("B_NuCI");
+    private By size = By.xpath("//*[text()='6']");
+    private By cart = By.xpath("//*[@class='_2KpZ6l _2U9uOA _3v1-ww']");
+    private By price = By.xpath("//*[@class='_30jeq3 _16Jk6d']");
+
+    public ProductPage(WebDriver driver) {
+        this.wait = new WebDriverWait(driver, 30);
+        this.driver = driver;
+    }
+
+    public ProductPage clickSize() {
+        wait.until(ExpectedConditions.elementToBeClickable(size));
+        driver.findElement(size).click();
+        return this;
+    }
+
+    public ProductPage addToCart() {
```

Comment:
should this return CartPage?

Reply:
It will just add to cart. wont go to cart page. next step is closing window.

Reply:
@dheerajsrinath But as per the actual flow - if you click on add to cart button -> it will navigate to Cart Page.
Agreed that in your test case, you are closing the tab, but if someone wants to reuse your code and wants to navigate to the cart page - then it will create a restriction. 
Please remember that its not only you who will be contributing to the repo.